core/property/registry.go

```diff
@@ -2,31 +2,66 @@ package property
 
 import (
 	"fmt"
-	"sync"
-
+	"github.com/TIBCOSoftware/flogo-lib/config"
 	"github.com/TIBCOSoftware/flogo-lib/core/data"
 	"github.com/TIBCOSoftware/flogo-lib/logger"
+	"reflect"
+	"regexp"
+	"sync"
 )
 
 var (
-	props = make(map[string]string)
-	mut   = sync.RWMutex{}
+	props    = make(map[string]interface{})
```

Comment:
As a framework, I expect OSS to load things as they are. 
This is same as any activity/trigger configuration stored and loaded by the OSS today. We should be consistent across all. Otherwise, It will confuse extension developers.
Since property types are pretty much defined(restricted) by the OSS, it makes more sense for OSS to do implicit casting than putting that burden on the extension developers.